### PR TITLE
V2 kernel processes

### PR DESCRIPTION
**Pull request by josiahhaswell**

### CODE REVIEW

File: kernel-api/src/main/java/io/sunshower/kernel/core/ValidationErrors.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  public boolean hasErrors() {
+    return !errors.isEmpty();
+  }
+
+  public static ValidationErrors empty() {
+    return new ValidationErrors();
+  }
+
+  public ValidationErrors merge(@NonNull ValidationErrors other) {
+    val result = new ArrayList<ValidationError>(errors.size() + other.errors.size());
```
Comment (lisagumerman) [inline kernel-api/src/main/java/io/sunshower/kernel/core/ValidationErrors.java:38]: What this?

File: kernel-api/src/main/java/io/sunshower/kernel/process/Phase.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Normal,
+
+    /** Warning: no action <i>strictly</i> required, but fix is desirable */
+    Warning,
+
+    /** Abnormal and unrecoverable (Kernel error) */
+    Error;
+
+    public static boolean canContinue(State state) {
+      return state == Normal || state == Recoverable;
```
Comment (lisagumerman) [inline kernel-api/src/main/java/io/sunshower/kernel/process/Phase.java:22]: Shouldn't this also be for Warning?

File: kernel-api/src/main/java/io/sunshower/kernel/status/StatusType.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package io.sunshower.kernel.status;
+
+public enum StatusType {
```
Comment (lisagumerman) [inline kernel-api/src/main/java/io/sunshower/kernel/status/StatusType.java:3]: This isn't parallel construction, but I guess we're also not in English class

File: kernel-core/src/main/java/io/sunshower/kernel/fs/FileSystemRegistry.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    public FileSystem next() {
+      while (!stack.empty()) {
+        var nextElement = stack.pop();
+        val children = nextElement.children;
+        stack.addAll(children);
+        if (nextElement.value != null) {
+          return nextElement.value;
+        }
+      }
+      throw new NoSuchElementException("Not here");
```
Comment (lisagumerman) [inline kernel-core/src/main/java/io/sunshower/kernel/fs/FileSystemRegistry.java:177]: Very helpful exception message. 🙄

File: kernel-core/src/main/java/io/sunshower/kernel/process/AbstractProcess.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  public T call() throws Exception {
+    for (val phase : phases) {
+      try {
+        phase.execute(this, context);
+      } catch (PhaseException ex) {
+        log.warning(ex.getMessage());
+        if (!Phase.State.canContinue(ex.getState())) {
+          return null;
+        } else {
+          //// TODO: 10/23/19 Add resolutions?
```
Comment (lisagumerman) [inline kernel-core/src/main/java/io/sunshower/kernel/process/AbstractProcess.java:65]: Still care about this TODO?

**Review by lisagumerman — CHANGES_REQUESTED** [review]

File: kernel-api/src/main/java/io/sunshower/kernel/core/ValidationErrors.java (hunk 1 above)
Comment (josiahhaswell) [reply kernel-api/src/main/java/io/sunshower/kernel/core/ValidationErrors.java:38]: Fixed in next PR

File: kernel-core/src/main/java/io/sunshower/kernel/process/AbstractProcess.java (hunk 5 above)
Comment (josiahhaswell) [reply kernel-core/src/main/java/io/sunshower/kernel/process/AbstractProcess.java:65]: I mean, it's a general todo--maybe next version

**Pull request merged**